iceoryx2-ffi/ffi/src/node.rs

```diff
@@ -92,6 +114,39 @@ impl iox2_node_storage_t {
     }
 }
 
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub enum iox2_node_state_e {
+    ALIVE,
+    DEAD,
+    INACCESSIBLE,
+    UNDEFINED,
+}
+
+pub type iox2_node_id_t = *const NodeId;
+
+/// An alias to a `void *` which can be used to pass arbitrary data to the callback
+pub type iox2_node_list_callback_ctx = *mut c_void;
```

Comment:
Style question: Should this be `iox2_node_list_cb_ctx` and `iox2_node_list_callback` further down `iox2_node_list_cb`

Reply:
I would prefer the longer names, even `iox2_node_list_callback_context` - it is less to think. But it is up to you, I see it as artistic freedom - feel free to add some emojis to transport the emotional meaning of the callback as well ;)